Library/ViewModels/SettingsAccountViewModel.swift

```diff
@@ -76,7 +79,17 @@ SettingsAccountViewModelOutputs, SettingsAccountViewModelType {
 
     self.reloadData = Signal.combineLatest(currency, shouldHideEmailWarning)
 
-    self.presentCurrencyPicker = currencyCellSelected.signal.mapConst(true).ignoreValues()
+    let updateCurrencyInProgress = Signal.merge(
+      self.viewDidLoadProperty.signal.mapConst(false),
+      didConfirmChangeCurrency.signal.mapConst(true),
+      updateCurrencyEvent.filter { $0.isTerminating }.mapConst(false)
+    )
+
+    self.presentCurrencyPicker = Signal.combineLatest(currency, updateCurrencyInProgress)
+      .takePairWhen(currencyCellSelected.signal)
+      .map { ($0.0, $0.1, $1) }
```

Comment:
I'd be happy to make this more readable if we find a better way

Reply:
Looks like this does compile, Xcode was just struggling before:

```swift
self.presentCurrencyPicker = Signal.combineLatest(currency, updateCurrencyInProgress)
  .takePairWhen(currencyCellSelected.signal)
  .map(unpack)
  .filter(second >>> isFalse)
  .map(first)
```